### PR TITLE
Adapt the readthedocs config to be similar to what ansible-lint is using

### PR DESCRIPTION
**Pull request by ziegenberg**
This commit changes the way the python dependencies for readthedocs are configured and it mirrors the way `ansible-lint` is doing it. The `docs` extra_requires is dropped and replaced with an `/docs/requirements.in` file. The `tox.ini` configuration is adapted to this change.

`sphinx.fail_on_warning` is set to `true` for readthedocs config. This turns warnings into errors (`-W` and `--keep-going options`). This means the build fails if there is a warning and exits with exit status 1. This mirrors the current configuration in `tox.ini`. So if building the docs locally fails, it now also fails with readthedocs.

A simple `Readme.md` for building documentation is added.

Also the pre-commit `pip-compile` steps are adapted, a `pip-compile-docs` and `pip-compile-docs-upgrade` hook is added.

Fixes: #3405

### CODE REVIEW

**Comment by ziegenberg** [comment]
@ssbarnea this is ready for review and it fixes building the docs at readthedocs.org  :)

**Comment by ziegenberg** [comment]
I'll try to come up with an alternative approach for this.

**Comment by ziegenberg** [comment]
I'll try to come up with an alternative approach for this.

**Comment by zollo** [comment]
I strongly suggest we get this in as soon as possible, the lack of documentation is making it difficult for us to use Molecule. Happy to lend my support in developing/reviewing this PR!

**Comment by zollo** [comment]
@ziegenberg thanks so much for your work on this!

File: requirements.txt [hunk 1]
```diff
@@ -2,15 +2,12 @@
 # This file is autogenerated by pip-compile with python 3.8
 # To update, run:
 #
-#    pip-compile --extra=docs --extra=test --extra=yamllint --no-annotate --output-file=requirements.txt --strip-extras setup.cfg tools/opts.txt
+#    pip-compile --extra=test --extra=yamllint --no-annotate --output-file=requirements.txt --strip-extras setup.cfg tools/opts.txt
```
Comment (ziegenberg) [inline requirements.txt:5]: @ssbarnea I guess this should be changed?
```suggestion
#    pip-compile --extra=test --extra=lint --no-annotate --output-file=requirements.txt --strip-extras setup.cfg tools/opts.txt
```

**Comment by ziegenberg** [comment]
py-devel keeps failing and I'm pretty sure I did not change anything related... :(

**Pull request closed without merging**